From 32d6a5224327b246ee3a2a08500544e4f80f9a9a Mon Sep 17 00:00:00 2001
From: Denis Chenu <denis@sondages.pro>
Date: Sun, 1 Sep 2019 16:27:10 +0200
Subject: [PATCH] Fixed issue: [security] #15204: Stored XSS vulnerabilities -
 Thanks to J. Greil from the SEC Consult Vulnerability Lab
 (https://www.sec-consult.com) Dev: encode group name in alert deleted

---
 application/controllers/admin/SurveysGroupsController.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/application/controllers/admin/SurveysGroupsController.php b/application/controllers/admin/SurveysGroupsController.php
index 31d971fe9a..2b97190ab5 100644
--- a/application/controllers/admin/SurveysGroupsController.php
+++ b/application/controllers/admin/SurveysGroupsController.php
@@ -130,7 +130,7 @@ public function delete($id)
 
             // if AJAX request (triggered by deletion via admin grid view), we should not redirect the browser
             if (!isset($_GET['ajax'])) {
-                Yii::app()->setFlashMessage(sprintf(gT("The survey group '%s' was deleted."), $sGroupTitle), 'success');
+                Yii::app()->setFlashMessage(sprintf(gT("The survey group '%s' was deleted."), CHtml::encode($sGroupTitle)), 'success');
                 $this->getController()->redirect(isset($_POST['returnUrl']) ? $_POST['returnUrl'] : array('admin/survey/sa/listsurveys '));
             }
         }
